isis/tests/CamptFunctionalTests.cpp

```diff
@@ -0,0 +1,139 @@
+    FAIL() << "Expected an exception to be thrown";
+  }
+  catch(Isis::IException &e) {
+    EXPECT_TRUE(e.toString().toLatin1().contains("Flat file must have a name."))
+      << e.toString().toStdString();
+  }
+  catch(...) {
+    FAIL() << "Expected an IException with message: \"Flat file must have a name.\"";
+  }
+}
+
+TEST_F(DefaultCube, FunctionalTestCamptDefaultParameters) {
+  QVector<QString> args = {};
+  UserInterface options(CAMPT_XML, args);
+  Pvl appLog;
+
+  campt(testCube, options, &appLog);
+  PvlGroup groundPoint = appLog.findGroup("GroundPoint");
+
+  EXPECT_DOUBLE_EQ( (double) groundPoint.findKeyword("Sample"), 602.0);
+  EXPECT_DOUBLE_EQ( (double) groundPoint.findKeyword("Line"), 528.0);
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, groundPoint.findKeyword("PixelValue"), "Null");
+  EXPECT_DOUBLE_EQ( (double) groundPoint.findKeyword("RightAscension"), 310.2070335306);
+  EXPECT_DOUBLE_EQ( (double) groundPoint.findKeyword("Declination"), -46.327246785573);
+  EXPECT_DOUBLE_EQ( (double) groundPoint.findKeyword("PlanetocentricLatitude"), 10.181441241544);
+  EXPECT_DOUBLE_EQ( (double) groundPoint.findKeyword("PlanetographicLatitude"), 10.299790241741);
+  EXPECT_DOUBLE_EQ( (double) groundPoint.findKeyword("PositiveEast360Longitude"), 255.89292858176);
+  EXPECT_DOUBLE_EQ( (double) groundPoint.findKeyword("PositiveEast180Longitude"), -104.10707141824);
+  EXPECT_DOUBLE_EQ( (double) groundPoint.findKeyword("PositiveWest360Longitude"), 104.10707141824);
+  EXPECT_DOUBLE_EQ( (double) groundPoint.findKeyword("PositiveWest180Longitude"), 104.10707141824);
+  
+  EXPECT_DOUBLE_EQ( toDouble(groundPoint.findKeyword("BodyFixedCoordinate")[0]), -818.59644749774);
+  EXPECT_DOUBLE_EQ( toDouble(groundPoint.findKeyword("BodyFixedCoordinate")[1]), -3257.2675597135);
+  EXPECT_DOUBLE_EQ( toDouble(groundPoint.findKeyword("BodyFixedCoordinate")[2]),  603.17640797124);
+
+  EXPECT_DOUBLE_EQ( (double) groundPoint.findKeyword("LocalRadius"), 3412288.6569795);
+  EXPECT_DOUBLE_EQ( (double) groundPoint.findKeyword("SampleResolution"), 18.904248467739);
+  EXPECT_DOUBLE_EQ( (double) groundPoint.findKeyword("LineResolution"), 18.904248467739);
+  EXPECT_DOUBLE_EQ( (double) groundPoint.findKeyword("ObliqueDetectorResolution"), 19.336214219327);
+  EXPECT_DOUBLE_EQ( (double) groundPoint.findKeyword("ObliquePixelResolution"), 19.336214219327);
+  EXPECT_DOUBLE_EQ( (double) groundPoint.findKeyword("ObliqueLineResolution"), 19.336214219327);
+  EXPECT_DOUBLE_EQ( (double) groundPoint.findKeyword("ObliqueSampleResolution"), 19.336214219327);
+
+  EXPECT_DOUBLE_EQ( toDouble(groundPoint.findKeyword("SpacecraftPosition")[0]), -1152.8979327717);
+  EXPECT_DOUBLE_EQ( toDouble(groundPoint.findKeyword("SpacecraftPosition")[1]), -3930.9421518203);
+  EXPECT_DOUBLE_EQ( toDouble(groundPoint.findKeyword("SpacecraftPosition")[2]),  728.14118380775);
+
+  EXPECT_DOUBLE_EQ( (double) groundPoint.findKeyword("SpacecraftAzimuth"), 240.08514246657);
```

Comment:
A few (5) of these `EXPECT_DOUBLE_EQ` lines are failing when I run them locally. The differences seem insignificant, for example: 
```
1679: Expected equality of these values:
1679:   (double) groundPoint.findKeyword("SpacecraftAzimuth")
1679:     Which is: 240.08514246636
1679:   240.08514246657
1679:     Which is: 240.08514246657001
```

Maybe we should use EXPECT_NEAR instead?

Reply:
Full list of failed EXPECT_DOUBLE_EQ's:
```
1679: Expected equality of these values:
1679:   (double) groundPoint.findKeyword("SpacecraftAzimuth")
1679:     Which is: 240.08514246636
1679:   240.08514246657
1679:     Which is: 240.08514246657001
1679: /home/pgiroux/repos/ISIS3/isis/tests/CamptFunctionalTests.cpp:113: Failure
1679: Expected equality of these values:
1679:   (double) groundPoint.findKeyword("SubSolarAzimuth")
1679:     Which is: 92.033828156504001
1679:   92.033828156965
1679:     Which is: 92.033828156965001
1679: /home/pgiroux/repos/ISIS3/isis/tests/CamptFunctionalTests.cpp:121: Failure
1679: Expected equality of these values:
1679:   (double) groundPoint.findKeyword("NorthAzimuth")
1679:     Which is: 332.65918494109002
1679:   332.65918493997
1679:     Which is: 332.65918493996998
1679: /home/pgiroux/repos/ISIS3/isis/tests/CamptFunctionalTests.cpp:136: Failure
1679: Expected equality of these values:
1679:   toDouble(groundPoint.findKeyword("LookDirectionCamera")[0])
1679:     Which is: -0.000127447324380692
1679:   -1.27447324380581e-04
1679:     Which is: -0.000127447324380581
1679: /home/pgiroux/repos/ISIS3/isis/tests/CamptFunctionalTests.cpp:137: Failure
1679: Expected equality of these values:
1679:   toDouble(groundPoint.findKeyword("LookDirectionCamera")[1])
1679:     Which is: 2.5816511718734801e-05
1679:   2.5816511718707e-05
1679:     Which is: 2.5816511718707001e-05
```